src/MessageTrait.php

```diff
@@ -400,4 +401,32 @@ private static function assertValidHeaderValue(array $values)
     {
         array_walk($values, __NAMESPACE__ . '\HeaderSecurity::assertValid');
     }
+
+    /**
+     * Validate the HTTP protocol version
+     *
+     * @param string $version
+     * @throws InvalidArgumentException on invalid HTTP protocol version
+     */
+    private function validateProtocolVersion($version)
+    {
+        if (empty($version)) {
+            throw new InvalidArgumentException(sprintf(
+                'HTTP protocol version can not be empty'
+            ));
+        }
+        if (! is_string($version)) {
+            throw new InvalidArgumentException(sprintf(
+                'Unsupported HTTP protocol version; must be a string, received %s',
+                (is_object($version) ? get_class($version) : gettype($version))
+            ));
+        }
+        //HTTP uses a "<major>.<minor>" numbering scheme to indicate versions of the protocol
+        if (! preg_match('~^\d+\.\d+$~', $version)) {
```

Comment:
HTTP/2 does not follow this format (see [the HTTP/2 FAQ](https://http2.github.io/faq/#is-it-http20-or-http2); no decimal point. I'll update this on merge.